src/append_only_zks.rs

```diff
@@ -130,7 +161,8 @@ impl Azks {
         append_only_usage: bool,
     ) -> Result<(), AkdError> {
         self.increment_epoch();
-
+        self.preload_nodes_for_insertion::<S, H>(storage, insertion_set.clone())
```

Comment:
Can we borrow the insertion set rather than cloning it? This set might be 100K+ items